test_embark/staking.js

```diff
@@ -0,0 +1,193 @@
+    token = tokenContract.methods
+    tokenAddress = tokenContract.options.address
+    await token.mint(other, defaultAmount).send()
+    const stakingContract = await StakingMock.deploy({arguments: [tokenAddress]}).send()
+    staking = stakingContract.methods
+    stakingAddress = stakingContract.options.address
+  })
+
+  it('has correct initial state', async () => {
+    assert.equal(await staking.token().call(), tokenAddress, "Token is wrong")
+    assert.equal((await staking.totalStaked().call()).valueOf(), 0, "Initial total staked amount should be zero")
+    assert.equal(await staking.supportsHistory().call(), true, "history support should match")
+  })
+
+  it('fails deploying if token is not a contract', async() => {
+    return assertRevert(async () => {
+      await StakingMock.deploy({arguments: [owner]}).send()
+    })
+  })
+
+  it('stakes', async () => {
+    const initialOwnerBalance = parseInt((await token.balanceOf(owner).call()).valueOf(), 10)
+    const initialStakingBalance = parseInt((await token.balanceOf(stakingAddress).call()).valueOf(), 10)
+
+    await approveAndStake()
+
+    const finalOwnerBalance = parseInt((await token.balanceOf(owner).call()).valueOf(), 10)
+    const finalStakingBalance = parseInt((await token.balanceOf(stakingAddress).call()).valueOf(), 10)
+    assert.equal(finalOwnerBalance, initialOwnerBalance - defaultAmount, "owner balance should match")
+    assert.equal(finalStakingBalance, initialStakingBalance + defaultAmount, "Staking app balance should match")
+    assert.equal((await staking.totalStakedFor(owner).call()).valueOf(), defaultAmount, "staked value should match")
+    // total stake
+    assert.equal((await staking.totalStaked().call()).toString(), defaultAmount, "Total stake should match")
+  })
+
+  it('fails staking 0 amount', async () => {
+    await token.approve(stakingAddress, 1).send()
+    return assertRevert(async () => {
+      await staking.stake(0, web3.utils.asciiToHex('')).send()
+    })
+  })
+
+  it('fails staking more than balance', async () => {
+    const balance = parseInt((await token.balanceOf(owner).call()).valueOf(), 10)
+    const amount = balance + 1
+    await token.approve(stakingAddress, amount).send()
+    return assertRevert(async () => {
+      await staking.stake(amount, web3.utils.asciiToHex('')).send()
+    })
+  })
+
+  it('stakes for', async () => {
+    const initialOwnerBalance = parseInt((await token.balanceOf(owner).call()).valueOf(), 10)
+    const initialOtherBalance = parseInt((await token.balanceOf(other).call()).valueOf(), 10)
+    const initialStakingBalance = parseInt((await token.balanceOf(stakingAddress).call()).valueOf(), 10)
+
+    // allow Staking app to move owner tokens
+    await token.approve(stakingAddress, defaultAmount).send()
+    // stake tokens
+    await staking.stakeFor(other, defaultAmount, web3.utils.asciiToHex('')).send()
+
+    const finalOwnerBalance = parseInt((await token.balanceOf(owner).call()).valueOf(), 10)
```

Comment:
These tests are a bit hard to follow given the separation of the call and the assertion. I just keep thinking that you are somehow comparing to `10` all the time.